### PR TITLE
Audit and proof API

### PR DESCRIPTION
**Pull request by Razikus**
Adds proof and audit API for objects storage

### CODE REVIEW

File: pkg/api/schemav2/schemav2.proto [hunk 1]
```diff
@@ -149,6 +170,26 @@ service ImmuServiceV2 {
...
+    option (google.api.http) = {
+      post: "/documents/audit"
+      body: "*"
+    };
+    option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {
+      tags: "documents";
+    };
+  } 
+ 
+  rpc DocumentProof(DocumentProofRequest) returns (schema.VerifiableTx) {
```
Comment (jeroiraz) [inline pkg/api/schemav2/schemav2.proto:184]: return type should be DocumentProofResponse

File: pkg/api/schemav2/schemav2.proto [hunk 2]
```diff
@@ -69,6 +69,31 @@ message DocumentSearchResponse {
...
 
+message DocumentProofRequest {
+  string collection = 1;
+  map<string, PossibleIndexType> primaryKeys = 2;
+  int64 atRevision = 3;
+}
+
+message DocumentAuditRequest {
+  string collection = 1;
+  map<string, PossibleIndexType> primaryKeys = 2;
```
Comment (jeroiraz) [inline pkg/api/schemav2/schemav2.proto:80]: not sure I understand the objective of `primaryKeys` here based on the defined type

File: pkg/api/schemav2/schemav2.proto [hunk 3]
```diff
@@ -69,6 +69,31 @@ message DocumentSearchResponse {
   uint32 entriesLeft = 4;
 }
 
+message DocumentProofRequest {
+  string collection = 1;
+  map<string, PossibleIndexType> primaryKeys = 2;
```
Comment (jeroiraz) [inline pkg/api/schemav2/schemav2.proto:74]: not sure I understand the objective of primaryKeys here based on the defined type

File: pkg/api/schemav2/schemav2.proto (hunk 1 above)
Comment (Razikus) [reply pkg/api/schemav2/schemav2.proto:184]: https://github.com/codenotary/immudb/pull/1546

File: pkg/api/schemav2/schemav2.proto (hunk 3 above)
Comment (Razikus) [reply pkg/api/schemav2/schemav2.proto:74]: https://github.com/codenotary/immudb/pull/1546

**Pull request closed without merging**